drivers/modem/gsm_ppp.c

```diff
@@ -290,18 +296,39 @@ static int gsm_setup_mccmno(struct gsm_modem *gsm)
 	return ret;
 }
 
+static struct net_if *ppp_net_if(void)
+{
+	return net_if_get_first_by_type(&NET_L2_GET_NAME(PPP));
+}
+
 static void set_ppp_carrier_on(struct gsm_modem *gsm)
 {
-	const struct device *ppp_dev = device_get_binding(CONFIG_NET_PPP_DRV_NAME);
-	const struct ppp_api *api;
+	const struct device *ppp_dev =
+		device_get_binding(CONFIG_NET_PPP_DRV_NAME);
+	static const struct ppp_api *api;
+	struct net_if *iface = ppp_net_if();
```

Comment:
So instead of this separate iface variable, we could do following
```
struct gsm_modem {
     ...
     struct net_if *iface;
     ...
};

static int gsm_init(...)
{
    ...
    gsm->iface = ppp_net_if();
    ....
}
```
then in this function we just refer to `gsm->iface` when the interface is needed. This follows the style how other drivers are doing it, and helps to isolate the interface setting to when init is done.